### ISSUE TITLE
Add config editing to the UI

### ISSUE THREAD

**Issue by JackDallas** [JackDallas/Premiumizearr#39]
As highlighted here https://github.com/JackDallas/Premiumizearr/issues/38 being able to edit the config from the ui is a very desired feature. There are a few ways to go about it though, from a security perspective:

- Implement a user/pass system to be set the first time the ui is opened (Most dev effort)
- Make properties write only so if the ui is exposed api keys can't be scraped (Medium dev effort)
- Warn people to stick this behind a reverse proxy (No dev effort)

At the moment my plan is simply a second page with the properties currently in the config listed and editable. If anyone has any suggestions or features they want this is the thread to post them in.

As I said all suggestions and such relating to this are welcome and I'd like to know how people handle security and which of these options they'd prefer

**Comment by Lloytron** [comment]
First of all I just wanted to say thanks for this project, its really exactly what I've been looking for for quite a long time now!

One suggestion, is it possible to add download management? I have some downloads that have completed but for some reason the program keeps on trying to redownload them and I can't remove them. I've deleted them and even wiped my whole docker environment but can't get rid of the damn things :D

But a mechanism to cancel a download would be really helpful. On top of this more general download management would be really good, such as;

- Pause/Resume
- Reordering the queue
- Throttling download speed (I have crap internet so when I'm downloading it locks everyone else out :D)

**Comment by JackDallas** [comment]
@Lloytron @rkershenbaum I've made an initial pre-release. 

Note this does not have authentication at all so needs to be either lan or behind a reverse proxy, I may have broke something along the way but it's working with my setup atm. Open to all kinds of feedback but know I'm no web designer lol. I've re wrote the config implementation so now the app will start up without the config being filled in, still working on making it so these values all can change without a restart but for now all the file based ones and web server ones need a restart.

Pre-Release:

https://github.com/JackDallas/Premiumizearr/releases/tag/v1.2-rc1

Docker is also pre-released (not tested):

https://github.com/jackdallas/Premiumizearr/pkgs/container/premiumizearr

(just noticed it's tagged it latest as well ffs, my pre-release workflow needs some work)

**Comment by Lloytron** [comment]
I've updated using the docker image but I dont see any difference?

BTW when I restart it starts downloading previous items that have already downloaded and I have no way to stop them even though the blackhole directory is empty. Is there any way I can delete queued items manually?

**Comment by JackDallas** [comment]
> BTW when I restart it starts downloading previous items that have already downloaded and I have no way to stop them even though the blackhole directory is empty. Is there any way I can delete queued items manually?

Sorry do you mean that under the Transfers or Downloads section,? Transfers is just what's happening on premiumize.me. Downloads automatically grabs any files in the `arrDownloads` folder on premiumize.me and then deletes the file from the cloud when it's complete so it's not re-downloaded

**Comment by JackDallas** [comment]
> [Further edit] I can confirm that the changes made in the UI successfully save to the config.yaml and vice versa so thats looking good.
> 
> However the config.yaml needs to be in place before deploying the docker container. If it's not then Docker makes a folder called config.yaml and the program throws an error. If you put a blank .yaml file there it also doesn't start so it needs to be a valid file already in place.
> 

@Lloytron Can you just not map the config at all now? Now it's all editable in the ui it can just act as a way to store not edit the config in docker. I'll change the README to match not doing that if it works.

Also fyi feel free to just reply back to the thread and not make big edits. that way it keeps a convo flow and I get notifications on gthub.

One config parameter is missing and I have a fix for a reverse proxy issue going up so I'll release that as an rc2 in a min

**Comment by Lloytron** [comment]
@JackDallas If I don't map the config file then the app doesn't start up and gives the following output in the logs;

time="2022-06-04T19:20:00Z" level=info msg="---------- Starting premiumizearr daemon ----------"
time="2022-06-04T19:20:00Z" level=info
time="2022-06-04T19:20:00Z" level=warning msg="No config file found, created default config file"
time="2022-06-04T19:20:00Z" level=warning msg="Premiumizeme API key not set, application will not work until it's set"
time="2022-06-04T19:20:00Z" level=info msg="Starting web server..."
time="2022-06-04T19:20:00Z" level=info msg="Starting directory watcher..."
time="2022-06-04T19:20:00Z" level=info msg="Web server started on :"
time="2022-06-04T19:20:02Z" level=info msg="Clearing tmp directory..."
time="2022-06-04T19:20:02Z" level=info msg="Creating Queue..."
time="2022-06-04T19:20:02Z" level=info msg="Starting uploads processor..."
time="2022-06-04T19:20:02Z" level=info msg="Starting initial directory scans..."
time="2022-06-04T19:20:02Z" level=error msg="Error with initial directory scan open : no such file or directory"

And yes I was having downloads restarting, thanks for explaining! I cleared the Arr download folder and thats sorted now. Cheers!

**Comment by JackDallas** [comment]
Oh crap I know exactly what this is, I'd forgotten, there's a bug, the first time starting up the web server will fail, also I've realised I'm being dumb, we need to be able to mount the config directory to keep it working, I'll sort both those and release rc3 as soon as I can :+1:

**Comment by JackDallas** [comment]
@Lloytron That issue is fixed, I've also set the Dockerfile up to initialise the config with the correct container side directories as shown in the README.

https://github.com/JackDallas/Premiumizearr/releases/tag/v1.2-rc5

Docker has uploaded, up to rc5 now but the latest tag is still being overwritten so you can just use that 

Updated README here (let me know if I got anything wrong):
https://github.com/JackDallas/Premiumizearr/tree/add-config-editing-ui#docker

**Comment by Lloytron** [comment]
@JackDallas Unfortunately something seems to have broken here and I can't get it to work at all now.

If I don't mount the data folder and just use the default then I get the following logs;

time="2022-06-05T16:53:06Z" level=info msg="---------- Starting premiumizearr daemon ----------"
time="2022-06-05T16:53:06Z" level=info
time="2022-06-05T16:53:06Z" level=warning msg="No config file found, created default config file"
time="2022-06-05T16:53:06Z" level=info msg="Starting directory watcher..."
time="2022-06-05T16:53:06Z" level=info msg="Starting web server..."
time="2022-06-05T16:53:06Z" level=info msg="Web server started on 0.0.0.0:8182"
time="2022-06-05T16:53:07Z" level=error msg="Error getting folders: error"
time="2022-06-05T16:53:07Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-05T16:53:07Z" level=error msg="Error getting folders: error"
time="2022-06-05T16:53:07Z" level=info msg="Clearing tmp directory..."
time="2022-06-05T16:53:07Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-05T16:53:08Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-05T16:53:10Z" level=error msg="Error clearing tmp directory /"
time="2022-06-05T16:53:10Z" level=info msg="Creating Queue..."
time="2022-06-05T16:53:10Z" level=info msg="Starting uploads processor..."
time="2022-06-05T16:53:10Z" level=info msg="Starting initial directory scans..."
time="2022-06-05T16:53:10Z" level=error msg="Error with initial directory scan open /blackhole: no such file or directory"
time="2022-06-05T16:53:23Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-05T16:53:38Z" level=error msg="Error listing downloads folder: Could not decode folder id"


If I do mount the data folder then the config.yaml file doesn't get created, although I do see log files appear!

**Comment by JackDallas** [comment]
Ohh the unzip directory apparently ending up as / will be deleting the config I bet 🤦‍♀️ I just wrote some code that will prevent that happening accidentally but that shouldn't happen in docker. I'm reworking the config so it's less tempremental rn, I'll see about sorting docker building locally as well so I can rest it better

**Comment by JRDevo** [comment]
> At the moment my plan is simply a second page with the properties currently in the config listed and editable. If anyone has any suggestions or features they want this is the thread to post them in.

It would be nice to have a test button next to each connection to ensure its working :) Or some simple status indicator.

**Comment by Lloytron** [comment]
Nice one, look forward to trying it!

BTW I just remembered that the config page is missing the option to add your Premiumize.me API key

**Comment by JackDallas** [comment]
> It would be nice to have a test button next to each connection to ensure its working :) Or some simple status indicator.

> BTW I just remembered that the config page is missing the option to add your Premiumize.me API key

Both great points, I added testing the arr clients and fixed the premiumize api key field. At this point everything can be saved on the config page and the settings will update without a restart. Unless anyone finds a bug in the next day or so I'll likely release this as 1.2 :)

Here's the release (also on docker):

https://github.com/JackDallas/Premiumizearr/releases/tag/v1.2-rc8

Let me know how it goes :+1:

**Comment by Lloytron** [comment]
I'm still getting errors if I don't mount the Data folder, or if there's no config.yaml file present when I do, however if the file is present then it works again, thanks :D

This is the log when the file isn't present; 

time="2022-06-05T22:55:00Z" level=info msg="---------- Starting premiumizearr daemon ----------"
time="2022-06-05T22:55:00Z" level=info
time="2022-06-05T22:55:00Z" level=warning msg="No config file found, created default config file"
time="2022-06-05T22:55:00Z" level=info msg="Starting web server..."
time="2022-06-05T22:55:00Z" level=info msg="Web server started on 0.0.0.0:8182"
time="2022-06-05T22:55:00Z" level=info msg="Starting directory watcher..."
time="2022-06-05T22:55:01Z" level=error msg="Error getting folders: error"
time="2022-06-05T22:55:01Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-05T22:55:01Z" level=info msg="Clearing tmp directory..."
panic: Unzip directory is set to system root don't do that it will wipe your system!

goroutine 1 [running]:
github.com/jackdallas/premiumizearr/internal/service.(*DirectoryWatcherService).Start(0x400016e2e8)
	/home/runner/work/Premiumizearr/Premiumizearr/internal/service/directory_watcher_service.go:71 +0x568
main.(*App).Start(0x400016e1c0, {0x42be37, 0x4}, {0x400002801e, 0x5}, {0x400002804f, 0x5})
	/home/runner/work/Premiumizearr/Premiumizearr/cmd/premiumizearrd/app.go:105 +0x914
main.main()
	/home/runner/work/Premiumizearr/Premiumizearr/cmd/premiumizearrd/main.go:22 +0x338
time="2022-06-05T22:55:04Z" level=info msg="---------- Starting premiumizearr daemon ----------"


Another issue I noticed (btW did you want us to raise these as seperate issues or just comment here?) is that when I restart the docker container everything in my mapped download volume gets deleted.

I had mapped
      - /home/pi/Media/Downloads/:/downloads
      
 And from this I have the blackhole and completed folders as /downloads/blackhole and /doesnloads/completed
 
 But when I restart the container these folder were being deleted which was causing errors in my *Arr applications which can't then find the folders to put stuff.

**Comment by JackDallas** [comment]
> Another issue I noticed (btW did you want us to raise these as seperate issues or just comment here?) i 
Here is perfect :+1:

I think those two issues are the same thing happening, can you post your whole docker run/create command here please

**Comment by Lloytron** [comment]
I'm using Docker Compose (via Portainer) to manage things. This is what I use in my compose file;

  premiumizearr:
    image: ghcr.io/jackdallas/premiumizearr:latest
    container_name: premiumizearr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - /home/pi/Media/Downloads:/downloads
      - /home/pi/Media/config/premiumizearr:/data
    network_mode: host
    restart: unless-stopped

**Comment by JRDevo** [comment]
Rather raise a new issue then use this one :)

**Comment by JackDallas** [comment]
@JRDevo Best to keep any issues with the 1.2-rc release's here just for my sanity (lol) weird the changes I made in rc10 fixed this for me but maybe I missed something.

Gonna pinch your docker command and test against it later 

> `sudo docker run -d --name=Premiumizearr -v /volumeUSB2/usbshare/config/premiumizearr/data:/data -e PUID=1033 -e PGID=100 -e TZ=Europe/Amsterdam -p 8182:8182 -v /volume1/Downloads:/downloads -v /volumeUSB2/usbshare/downloads/blackhole:/blackhole --restart unless-stopped ghcr.io/jackdallas/premiumizearr:latest`

You mentioned `Similarly with ghcr.io/jackdallas/premiumizearr:v1.2-rc10` can I just confirm you tried the above command with that version instead of latest

**Comment by Lloytron** [comment]
If I understand correctly the config,yaml file location in the container isn't /opt/app any more but is in the data folder?

**Comment by JackDallas** [comment]
> If I understand correctly the config,yaml file location in the container isn't /opt/app any more but is in the data folder?
That's correct 

@JRDevo I wouldn't expect the first docker command that maps to `/opt/app/config.yaml` to work any more because of these changes 

The second command looks to me like it should work, are you getting the same `panic: Unzip directory is set to system root don't do that it will wipe your system!` error ?

**Comment by JackDallas** [comment]
@Lloytron 

> I'm using Docker Compose (via Portainer) to manage things. This is what I use in my compose file;
> 
> premiumizearr: image: ghcr.io/jackdallas/premiumizearr:latest container_name: premiumizearr environment: - PUID=1000 - PGID=1000 - TZ=Europe/London volumes: - /home/pi/Media/Downloads:/downloads - /home/pi/Media/config/premiumizearr:/data network_mode: host restart: unless-stopped

This should work with the latest rc10 I think.

**Comment by JackDallas** [comment]
I've realised the issue with the config not appearing, until you make a change in the ui and click save nothing is wrote to disk, I have this replicated in docker but weirdly not on bare metal so I'm working on debugging in docker now

**Comment by JRDevo** [comment]
> If I understand correctly the config,yaml file location in the container isn't /opt/app any more but is in the data folder?

> @JRDevo I wouldn't expect the first docker command that maps to `/opt/app/config.yaml` to work any more because of these changes

Weirdly yes, but this time Ive emptied the data folder and seems to be working with:

`sudo docker run -d   --name=Premiumizearr   -v /volumeUSB2/usbshare/config/premiumizearr/data:/data   -e PUID=1033   -e PGID=100   -e TZ=Europe/Amsterdam   -p 8182:8182   -v /volume1/Downloads:/downloads   -v /volumeUSB2/usbshare/downloads/blackhole:/blackhole   --restart unless-stopped   ghcr.io/jackdallas/premiumizearr:v1.2-rc10`

It looks like the new fixes are working then!

The new config window looking good :)

**Comment by Lloytron** [comment]
> @Lloytron
> 
> > I'm using Docker Compose (via Portainer) to manage things. This is what I use in my compose file;
> > premiumizearr: image: ghcr.io/jackdallas/premiumizearr:latest container_name: premiumizearr environment: - PUID=1000 - PGID=1000 - TZ=Europe/London volumes: - /home/pi/Media/Downloads:/downloads - /home/pi/Media/config/premiumizearr:/data network_mode: host restart: unless-stopped
> 
> This should work with the latest rc10 I think.

I just tried rc10 and its the same, seems to give the same error now regardless of how the data folder is mapped!

**Comment by JRDevo** [comment]
> > @Lloytron
> > > I'm using Docker Compose (via Portainer) to manage things. This is what I use in my compose file;
> > > premiumizearr: image: ghcr.io/jackdallas/premiumizearr:latest container_name: premiumizearr environment: - PUID=1000 - PGID=1000 - TZ=Europe/London volumes: - /home/pi/Media/Downloads:/downloads - /home/pi/Media/config/premiumizearr:/data network_mode: host restart: unless-stopped
> > 
> > 
> > This should work with the latest rc10 I think.
> 
> I just tried rc10 and its the same, seems to give the same error now regardless of how the data folder is mapped!

Yeah with @Lloytron again here.

It worked once for me, then i deleted the container, image and cleared the folder and ran it again:

`sudo docker run -d   --name=Premiumizearr   -v /volumeUSB2/usbshare/config/premiumizearr/data:/data   -e PUID=1033   -e PGID=100   -e TZ=Europe/Amsterdam   -p 8182:8182   -v /volume1/Downloads:/downloads   -v /volumeUSB2/usbshare/downloads/blackhole:/blackhole   --restart unless-stopped   ghcr.io/jackdallas/premiumizearr:v1.2-rc10`

```
Unable to find image 'ghcr.io/jackdallas/premiumizearr:v1.2-rc10' locally
v1.2-rc10: Pulling from jackdallas/premiumizearr
125a6e411906: Pull complete
625f7960341a: Pull complete
ff484277ac4b: Pull complete
497801505c02: Pull complete
eb2e5016aded: Pull complete
5dff818df92d: Pull complete
e8976f4bdfe7: Pull complete
d02dbd399c86: Pull complete
523ca4f6f854: Pull complete
Digest: sha256:5d8afec73ef400666b53f10a518e1aaf420c5a6ff4a5300e0c3044176f76c6fa
Status: Downloaded newer image for ghcr.io/jackdallas/premiumizearr:v1.2-rc10
0c0b6f128bd2e2c69ba77c40a485ba7f727a2b33f36c7adbf9032b894ffca848
```

Error: 

```
time="2022-06-06T12:57:22Z" level=info msg="---------- Starting premiumizearr daemon ----------"
time="2022-06-06T12:57:22Z" level=info
time="2022-06-06T12:57:22Z" level=warning msg="No config file found, created default config file"
time="2022-06-06T12:57:22Z" level=info msg="Starting web server..."
time="2022-06-06T12:57:22Z" level=info msg="Web server started on 0.0.0.0:8182"
time="2022-06-06T12:57:22Z" level=info msg="Starting directory watcher..."
time="2022-06-06T12:57:22Z" level=error msg="Error getting folders: error"
time="2022-06-06T12:57:22Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-06T12:57:22Z" level=info msg="Clearing tmp directory..."
panic: Unzip directory is set to system root don't do that it will wipe your system!
goroutine 1 [running]:
github.com/jackdallas/premiumizearr/internal/service.(*DirectoryWatcherService).Start(0xc0000c02e8)
	/home/runner/work/Premiumizearr/Premiumizearr/internal/service/directory_watcher_service.go:71 +0x574
main.(*App).Start(0xc0000c01c0, {0x8327ce, 0xc00001ef80}, {0xc00002601e, 0x5}, {0xc00002604f, 0x5})
	/home/runner/work/Premiumizearr/Premiumizearr/cmd/premiumizearrd/app.go:105 +0xa46
main.main()
	/home/runner/work/Premiumizearr/Premiumizearr/cmd/premiumizearrd/main.go:22 +0x235
```

**Comment by JackDallas** [comment]
Okay I think I got it !

rc-11 out now on docker. this works for me if y'all could add ` -e PREMIUMIZEARR_LOG_LEVEL=trace ` to your command if you still get errors and reply back with the log, or just let me know it works I'd appreciate it 

```cmd
dallas@cartman:~$ sudo rm -r test/data/*
[sudo] password for dallas:
dallas@cartman:~$ docker run -v /home/dallas/test/data:/data -v /home/dallas/test/blackhole:/blackhole -v /home/dallas/test/downloads:/downloads -p 8182:8182 -e PREMIUMIZEARR_LOG_LEVEL=trace ghcr.io/jackdallas/premiumizearr:v1.2-rc11
Unable to find image 'ghcr.io/jackdallas/premiumizearr:v1.2-rc11' locally
v1.2-rc11: Pulling from jackdallas/premiumizearr
125a6e411906: Already exists
22f185bc5a16: Pull complete
6796777e0bba: Pull complete
55047dd12baa: Pull complete
c0d03191bb5b: Pull complete
4632378dd890: Pull complete
11bb6ff33f65: Pull complete
e09e2ffab0f0: Pull complete
c3c20f846b9a: Pull complete
Digest: sha256:93a0397f6c5b4f5a18760013b19dcbd45c5d1ea6e500c164e2c7c0929ae06ba4
Status: Downloaded newer image for ghcr.io/jackdallas/premiumizearr:v1.2-rc11
time="2022-06-06T15:08:44Z" level=info msg="---------- Starting premiumizearr daemon ----------"
time="2022-06-06T15:08:44Z" level=info
time="2022-06-06T15:08:44Z" level=trace msg="Running load or create config"
time="2022-06-06T15:08:44Z" level=trace msg="Reading config file location from flag or env: /data"
time="2022-06-06T15:08:44Z" level=trace msg="Trying to load config from disk"
time="2022-06-06T15:08:44Z" level=trace msg="Reading config from /data/config.yaml"
time="2022-06-06T15:08:44Z" level=trace msg="Failed to find config file"
time="2022-06-06T15:08:44Z" level=warning msg="No config file found, created default config file"
time="2022-06-06T15:08:44Z" level=info msg="Running in docker, overriding unzip directory!"
time="2022-06-06T15:08:44Z" level=trace msg="Running in docker, overriding blank directory settings for blackhole directory"
time="2022-06-06T15:08:44Z" level=trace msg="Running in docker, overriding blank directory settings for downloads directory"
time="2022-06-06T15:08:44Z" level=trace msg="Setting config location to /data"
time="2022-06-06T15:08:44Z" level=trace msg="Marshaling & saving config"
time="2022-06-06T15:08:44Z" level=trace msg="Writing config to /data/config.yaml"
time="2022-06-06T15:08:44Z" level=trace msg="Config saved"
time="2022-06-06T15:08:44Z" level=info msg="Cleaning unzip directory"
time="2022-06-06T15:08:44Z" level=trace msg="Unzip directory set to: /unzip"
time="2022-06-06T15:08:44Z" level=info msg="Starting web server..."
time="2022-06-06T15:08:44Z" level=info msg="Web server started on 0.0.0.0:8182"
time="2022-06-06T15:08:44Z" level=info msg="Starting directory watcher..."
time="2022-06-06T15:08:44Z" level=trace msg="Getting folder list from premiumize.me"
time="2022-06-06T15:08:44Z" level=error msg="Error getting folders: error"
time="2022-06-06T15:08:44Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-06T15:08:44Z" level=info msg="Creating Queue..."
time="2022-06-06T15:08:44Z" level=info msg="Starting uploads processor..."
time="2022-06-06T15:08:44Z" level=info msg="Running initial directory scan..."
time="2022-06-06T15:08:44Z" level=trace msg="Getting folder list from premiumize.me"
time="2022-06-06T15:08:44Z" level=trace msg="No files in Queue, sleeping for 10 seconds"
time="2022-06-06T15:08:44Z" level=trace msg="Initial directory scan"
time="2022-06-06T15:08:44Z" level=error msg="Error getting folders: error"
time="2022-06-06T15:08:44Z" level=error msg="Cannot read folders from premiumize.me, application will not run!"
time="2022-06-06T15:08:44Z" level=debug msg="Running Task UpdateTransfersList"
time="2022-06-06T15:08:44Z" level=trace msg="Getting transfers list from premiumize.me"
time="2022-06-06T15:08:45Z" level=trace msg="Received 0 transfers"
time="2022-06-06T15:08:45Z" level=debug msg="Running Task CheckPremiumizeDownloadsFolder"
time="2022-06-06T15:08:45Z" level=trace msg="Reading response"
time="2022-06-06T15:08:45Z" level=error msg="Error listing downloads folder: Could not decode folder id"
^Cdallas@cartman:~$ ls test/data/
config.yaml  premiumizearr.error.log  premiumizearr.general.log  premiumizearr.info.log
dallas@cartman:~$ cat test/data/config.yaml
PremiumizemeAPIKey: xxxxxxxxx
Arrs:
- Name: Sonarr
  URL: http://localhost:8989
  APIKey: xxxxxxxxx
  Type: Sonarr
- Name: Radarr
  URL: http://localhost:7878
  APIKey: xxxxxxxxx
  Type: Radarr
BlackholeDirectory: /blackhole
DownloadsDirectory: /downloads
UnzipDirectory: /unzip
bindIP: 0.0.0.0
bindPort: "8182"
WebRoot: ""
SimultaneousDownloads: 5
```

**Comment by Lloytron** [comment]
I just updated the docker container and it looks good. UI appears and config.yaml file was created on my remote drive with default values.

**Comment by JackDallas** [comment]
> Working like a charm thanks @JackDallas!

Excellent news, once @Lloytron confirms their setup also works I'll merge this, tag 1.2 and start looking at refactoring the downloads manager and fixing docker overriding latest with rc's 🙄

**Comment by JRDevo** [comment]
Not sure if this is concerning? 

```
time="2022-06-06T16:43:57Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:44:12Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:44:27Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:44:42Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:44:57Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:45:13Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:45:28Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:45:43Z" level=error msg="Error listing downloads folder: Could not decode folder id"
time="2022-06-06T16:45:58Z" level=error msg="Error listing downloads folder: Could not decode folder id"
```

**Comment by JRDevo** [comment]
yeah, if I reload the config page then all the settings remain same if I restart the container

**Comment by JRDevo** [comment]
think deleting the folder in premiumize.me resolved this

**Comment by JackDallas** [comment]
Now that's odd ... I'm gonna ignore that for the moment if other people report it as an issue I'll have a look at handling it better. I will look at making that error message more clear before the 1.2 release though

**Comment by JackDallas** [comment]
I've merged in the changes and am tagging v1.2 now. Thanks for all the help @Lloytron & @JRDevo you've made this a lot quicker. I'm gonna get the pre-releases not overriding the docker latest tag next then I'll start work on the downloads manager changes https://github.com/JackDallas/Premiumizearr/issues/41
